modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/integration/OperatorsExtensionIntegrationTest.java

```diff
@@ -0,0 +1,160 @@
+import org.apache.calcite.adapter.enumerable.NullPolicy;
+import org.apache.calcite.avatica.util.TimeUnitRange;
+import org.apache.calcite.rex.RexBuilder;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlFunction;
+import org.apache.calcite.sql.SqlFunctionCategory;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.sql.fun.SqlTrimFunction;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.sql.type.OperandTypes;
+import org.apache.calcite.sql.type.ReturnTypes;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.apache.calcite.sql.validate.SqlValidator;
+import org.apache.calcite.sql2rel.SqlRexContext;
+import org.apache.calcite.sql2rel.SqlRexConvertlet;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.processors.query.calcite.exec.exp.RexImpTable;
+import org.apache.ignite.internal.processors.query.calcite.prepare.IgniteConvertletTable;
+import org.apache.ignite.internal.processors.query.calcite.prepare.IgniteSqlCallRewriteTable;
+import org.apache.ignite.internal.processors.query.calcite.sql.fun.IgniteOwnSqlOperatorTable;
+import org.apache.ignite.plugin.AbstractTestPluginProvider;
+import org.apache.ignite.plugin.PluginContext;
+import org.junit.Test;
+
+/** */
+public class OperatorsExtensionIntegrationTest extends AbstractBasicIntegrationTest {
+    /** */
+    private static final SqlFunction SUBSTR =
+        new SqlFunction(
+            "SUBSTR",
+            SqlKind.OTHER_FUNCTION,
+            ReturnTypes.ARG0_NULLABLE_VARYING,
+            null,
+            OperandTypes.STRING_INTEGER_INTEGER,
+            SqlFunctionCategory.STRING);
+
+    /** */
+    private static final SqlFunction TRUNC =
+        new SqlFunction(
+            "TRUNC",
+            SqlKind.OTHER_FUNCTION,
+            ReturnTypes.ARG0_OR_EXACT_NO_SCALE,
+            null,
+            OperandTypes.TIMESTAMP,
+            SqlFunctionCategory.TIMEDATE);
+
+    /** */
+    private static final SqlFunction TO_NUMBER =
+        new SqlFunction(
+            "TO_NUMBER",
+            SqlKind.OTHER_FUNCTION,
+            opBinding -> opBinding.getTypeFactory().createSqlType(SqlTypeName.DECIMAL),
+            null,
+            OperandTypes.STRING,
+            SqlFunctionCategory.NUMERIC);
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        return super.getConfiguration(igniteInstanceName)
+            .setPluginProviders(new AbstractTestPluginProvider() {
+                @Override public String name() {
+                    return "Test operators extension plugin";
+                }
+
+                @Override public void start(PluginContext ctx) {
+                    IgniteOwnSqlOperatorTable.instance().register(SUBSTR);
```

Comment:
If we want to extend a default operator, `IgniteOwnSqlOperatorTable` should go first in `FrameworkConfig` (`CalciteQueryProcessor#FRAMEWORK_CONFIG`). Otherwise, filtering operator overload by strict operands types isn't reliable. Operand types might not be consedered in `SqlUtil#filterRoutinesByParameterTypeAndName()`